modules/augment/src/transform.cpp

```diff
@@ -103,19 +102,17 @@ void Transform::rectangles(InputArray _src, OutputArray _dst)
 
 }
 
-std::vector<Mat> Transform::polygons(std::vector<Mat> src)
+void Transform::polygons(std::vector<Mat> src, OutputArrayOfArrays dst)
 {
-    std::vector<Mat> dst(src.size());
+    dst.create(src.size(), 1, 0, -1, true);
```

Comment:
What the meaning of these arguments? `dst` is a vector of cv::Mat. Please check how in OpenCV's source code `OutputArrayOfArrays` is used.

Reply:
This is the only way to create vector of Mat inside OuputArrayOfArrays.. it is used the same way in findContours().. the first is the rows and cols=1 because it is a 1d vector of Mat and -1 means it is the root vector and the last is allowTransposed.

It must be used this way to avoid adding data to vector<Mat> and copying it to the output array in the end which would require twice the storage temporarly in ram

Reply:
1) OutputArrayOfArrays might be already allocated. In example from the previous call
2) At this point we don't know the sizes of output Mat

Reply:
1. and it might not be allocated.. the documentation of _OutputArray which is the base class for OutputArrayOfArrays states explicitly that each function must call create on the _OutputArray before using it

2. we know the size of the output mat because the output polygon will have the same number of vertices as the input graph